requests/configuration/set.go

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ *******************************************************************************/
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+type Key string
+
+/*
+ * SetConfigRequest is for SMA processing of incoming (PUT) requests to Set Config.
+ *
+ * SetConfigRequest and SetConfigResponse struct each follows
+ */
+type SetConfigRequest struct {
+	Key         string `bson:"key" json:"key,omitempty"`
+	Value       string `bson:"value" json:"value,omitempty"`
+	isValidated bool   // internal member used for validation check
+}
+
+//Implements unmarshaling of JSON string to SetConfigRequest type instance
+func (sc *SetConfigRequest) UnmarshalJSON(data []byte) error {
+	var err error
+	test := struct {
+		Key   *string `json:"key"`
+		Value *string `json:"value"`
+	}{}
+
+	//Verify that incoming string will unmarshal successfully
+	if err = json.Unmarshal(data, &test); err != nil {
+		return err
+	}
+
+	sc.isValidated, err = sc.Validate()
```

Comment:
`Validate()` should be called after the properties on `sc` are set. Otherwise the instance is effectively empty.

Example
https://github.com/edgexfoundry/go-mod-core-contracts/blob/4a4a40fd862f5e9270a97718ee09393586d09a2a/models/device.go#L127

Also, note how `err` is returned in that example. For the code here, the `err` being returned from the call to `Validate()` is lost.

Reply:
Excellent feedback here and in the remainder of the PR as well, @tsconn23 - Thanks, I'm on it.